tests/test_cases/test_cocotb/test_scheduler.py

```diff
@@ -426,3 +441,27 @@ async def example():
     assert a == 0
     await NullTrigger()
     assert a == 1
+
+
+@cocotb.test()
+async def test_start_soon_scheduling(_):
+    async def coro():
+        nonlocal start_time
+        assert cocotb.utils.get_sim_time() == start_time
+
+    start_time = cocotb.utils.get_sim_time()
+    cocotb.scheduler.start_soon(coro())
+    await NullTrigger()
+    assert cocotb.utils.get_sim_time() == start_time
```

Comment:
This is the test I was talking about. Since there is no passage of time anywhere, `get_sim_time` isn't a good value. If we are just testing to ensure the task hasn't been started until after we yield control, there is an easier way.
```suggestion
    async def coro():
        pass

    task = cocotb.scheduler.start_soon(coro())
    assert not task._started
    await NullTrigger()
    assert task._started
```

Reply:
I wanted to test that `coro` started in the same time step.

Reply:
What I'm saying is that it's impossible not to, even if there was a bug, since there are no triggers which advance simulation time. There is no chance for incorrect behavior to occur.

Reply:
Also, time step might not be good enough, we don't want it to go back to the simulator at all before servicing. `get_sim_time` will be the same after a `ReadWrite`, for example.

Reply:
Figured out something by wrapping `_react` and checking before returning to simulator.